packages/api/src/models/revenueTypeModel.js

```diff
@@ -42,6 +42,8 @@ class RevenueType extends baseModel {
         revenue_name: { type: 'string', minLength: 1, maxLength: 100 },
         farm_id: { type: 'string' },
         revenue_translation_key: { type: 'string' },
+        agriculture_associated: { type: 'boolean' },
```

Comment:
Bringing up again the required property. I think we want required because we are then saying to developers/api users that these are "optional" but not giving users the opportunity to make it optional on the frontend.

Reply:
Ok let's make it required on the model (to also cover API users), but keep default values on the database level. From what I read on the StackOverflow links you shared, that should be fine. I took advantage of the default database inserts to keep more tests running without modification (and as a bonus they also keep the custom types already on beta or local behaving as before so I quite like them). But I agree the application level requirements should be the same via API or frontend so it belongs on the model.

Reply:
Yeah I was thinking I was glad this was not live because it would be hard to update the user data

Reply:
@Duncan-Brain When coding it up I realized that we do not require `crop_generated` on the frontend (if `agriculture_associated` is false on the frontend nothing is sent for `crop_generated` and the database defaults it to false). So I matched the model requirements to that so API and front behave the same.